src/main/java/uk/gov/hmcts/ccd/data/roleassignment/DefaultRoleAssignmentRepository.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.ccd.data.roleassignment;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.hmcts.ccd.ApplicationParams;
+import uk.gov.hmcts.ccd.data.SecurityUtils;
+import uk.gov.hmcts.ccd.data.definition.DefaultCaseDefinitionRepository;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ResourceNotFoundException;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ServiceException;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+@Qualifier(DefaultRoleAssignmentRepository.QUALIFIER)
+public class DefaultRoleAssignmentRepository implements RoleAssignmentRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultRoleAssignmentRepository.class);
+
+    public static final String QUALIFIER = "default";
+    public static final String AM_ERROR = "Problem getting Role Assignments from AccessManagement store because of %s";
+
+    private final ApplicationParams applicationParams;
+    private final SecurityUtils securityUtils;
+    private final RestTemplate restTemplate;
+
+    public DefaultRoleAssignmentRepository(final ApplicationParams applicationParams,
```

Comment:
is it possible to write test cases for this class

Reply:
I agree. We agreed with Rafal not to write tests for now for classes that are very likely to change. But this one I think it's going to stay unchanged I'm sure, so we should definitely write all tests for it